benchmark/inference/inference_benchmark.py

```diff
@@ -0,0 +1,128 @@
+                print(f'Configuration of {dataset_name} + {model_name} '
+                      f'not supported. Skipping.')
+                continue
+            print(f'Evaluation bench for {model_name}:')
+            if model_name == 'pna_conv':
+                loader = NeighborLoader(
+                    copy.copy(data),
+                    num_neighbors=[-1],
+                    input_nodes=mask,
+                    batch_size=1024,
+                    shuffle=False,
+                    num_workers=args.num_workers,
+                )
+                degree = get_degree(loader)
+
+            for batch_size in args.eval_batch_sizes:
+                subgraph_loader = NeighborLoader(
+                    copy.copy(data),
+                    num_neighbors=[-1],
+                    input_nodes=mask,
+                    batch_size=batch_size,
+                    shuffle=False,
+                    num_workers=args.num_workers,
+                )
+                subgraph_loader.data.n_id = torch.arange(data.num_nodes)
+
+                for layers in args.num_layers:
+                    for hidden_channels in args.num_hidden_channels:
+                        print(
+                            '-----------------------------------------------')
+                        print(f'Batch size={batch_size}, '
+                              f'Layers amount={layers}, '
+                              f'Hidden features size={hidden_channels}')
+                        params = {
+                            'inputs_channels': inputs_channels,
+                            'hidden_channels': hidden_channels,
+                            'output_channels': dataset.num_classes,
+                            'num_heads': args.num_heads,
+                            'num_layers': layers,
+                        }
+                        if model_name == 'pna_conv':
+                            params['degree'] = degree
+
+                        model = get_model(
+                            model_name, params, metadata=data.metadata()
+                            if dataset_name == 'ogbn-mag' else None)
+
+                        if args.pure_gnn_mode:
+                            prebatched_samples = []
+                            for i, batch in enumerate(subgraph_loader):
+                                if i == args.prebatched_samples:
+                                    break
+                                prebatched_samples.append(batch)
+                            subgraph_loader = prebatched_samples
+
+                        start = default_timer()
+                        model.inference(subgraph_loader, args.device)
+                        stop = default_timer()
+                        print(f'Inference time={stop-start:.3f}\n')
+
+
+if __name__ == '__main__':
+    argparser = argparse.ArgumentParser('GNN inference benchmark')
+
+    argparser.add_argument('--device', default='cpu', type=str)
```

Comment:
Could you please remove this `--device` arg, and use `torch.cuda.is_available()` to decide the device? It can align with other benchmarks, and reduce one arg. Meanwhile, model is also need to `to(device)`.